templates/addons/s3/outputs.yml

```diff
@@ -0,0 +1,6 @@
+{{.BucketResourceName}}BucketName:
+  Description: "The name of a user-defined bucket."
+  Value: !Sub ${App}-${Env}-${Service}-{{.BucketName}}
```

Comment:
Let's use `!Ref {{.BucketResourceName}}` so customers can change the name in just one place.

Reply:
I originally thought to use the Go template parser functions to fill in all values in {{.DoubleBraces}} and sanitize in the template package. BucketResourceName will be a logical ID sanitized version of BucketName.

Reply:
BucketName is the user-specified and validated s3 bucket name. BucketResourceName is the logical ID constructed from BucketName. It's difficult because s3 Buckets can contain dashes and dots, but ddb table names can contain dashes, dots, and underscores. So we need a more robust Logical ID sanitizer than the one in the cloudformation package.